### PR TITLE
feat(suite): NumberInput improvements

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* add undo and redo functionality to the NumberInput component
* fix formatting bug

## Related Issue

https://github.com/trezor/trezor-suite/issues/8067

### CODE REVIEW

**Comment by komret** [comment]
A whole number turns into decimal number when it is too long:
chrome-capture-2023-4-4

File: packages/suite/src/components/suite/NumberInput.tsx [hunk 1]
```diff
@@ -104,7 +101,12 @@ export const NumberInput = ({
         defaultValue,
     });
 
+    const locale = useSelector(state => state.suite.settings.language);
+    const [pressedKey, setPressedKey] = useState('');
     const [displayValue, setDisplayValue] = useState(localizeNumber(value || '', locale));
+    const initialHistoryValue = value ? [value] : [''];
```
Comment (komret) [inline packages/suite/src/components/suite/NumberInput.tsx]: Will this work correctly if `value` is `0`? Same for the line above. I think a better approach might be to provide a default value for `value` on line 95 (`value = ''`) and remove the fallbacks from lines 106 and 107.

**Review by komret — CHANGES_REQUESTED** [review]

**Comment by komret** [comment]
There are some differences to how undo/redo behaves by default:

1. Undo/redo actions appear when you right-click the field, but do not work properly.
2. Undo/redo is form specific, i.e. CMD+Z in field A will undo changes in field B if they are the most recent. This is ignored by `preventDefault` in the numeric fields, but it will still attempt to undo changes in numeric fields from other inputs, which again does not work properly.
3. Undo/redo history also remembers cursor position and moves it to where it was when the changes happened - numeric inputs do not do this.

Not sure if we can do anything about points 1 and 2, but maybe 3 can be implemented. What do you think?

**Comment by dahaca** [comment]
@sime Should we even allow to enter unreasonably large numbers? Or perhaps it's a good idea to limit entry.

File: packages/suite/src/components/suite/NumberInput.tsx (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/components/suite/NumberInput.tsx]: f52f6b73d

**Comment by sime** [comment]
I don't know what the upper limits can be.

Is it an issue if allow big numbers? @dahaca

**Comment by komret** [comment]
Looks like large numbers are working after f52f6b73d69327847ce70181824c6d8a74ac7ecc, but I found an issue with long decimals - I cannot get past certain length. Not that big of a deal as we usually have a validation limit on fractional length, but still a bit weird. Also, when I am in this state, `value` can differ from `displayValue`:
chrome-capture-2023-4-4 (1)

**Review by komret — APPROVED** [review]
Some important fixes and an addition of undo/redo functionality.

Undo/redo has some flaws, but we decided not to worry about them now, see [comment above](https://github.com/trezor/trezor-suite/pull/8349#issuecomment-1534721716).

There is a bug in `localizeNumber` util, which affects this component and will be dealt with in another issue:  #8391.

**Pull request merged**